tests/page/page-route.spec.ts

```diff
@@ -837,3 +837,76 @@ for (const method of ['fulfill', 'continue', 'abort'] as const) {
     expect(e.message).toContain('Route is already handled!');
   });
 }
+
+it('should chain continue', async ({ page, server }) => {
+  const intercepted = [];
+  await page.route('**/empty.html', route => {
+    intercepted.push(1);
+    route.continue();
+  });
+  await page.route('**/empty.html', route => {
+    intercepted.push(2);
+    route.continue();
+  });
+  await page.route('**/empty.html', route => {
+    intercepted.push(3);
+    route.continue();
+  });
+  await page.goto(server.EMPTY_PAGE);
+  expect(intercepted).toEqual([3, 2, 1]);
+});
+
+it('should not chain fulfill', async ({ page, server }) => {
+  await page.route('**/empty.html', route => {
+    route.continue();
+  });
+  await page.route('**/empty.html', route => {
+    route.fulfill({ status: 200, body: 'fulfilled' });
+  });
+  await page.route('**/empty.html', route => {
+    route.continue();
+  });
+  const response = await page.goto(server.EMPTY_PAGE);
+  const body = await response.body();
+  expect(body.toString()).toEqual('fulfilled');
+});
+
+it('should not chain abort', async ({ page, server }) => {
+  await page.route('**/empty.html', route => {
+    route.continue();
+  });
+  await page.route('**/empty.html', route => {
+    route.abort();
+  });
+  await page.route('**/empty.html', route => {
+    route.continue();
+  });
+  const e = await page.goto(server.EMPTY_PAGE).catch(e => e);
+  expect(e).toBeTruthy();
+});
+
+it('should continue after exception', async ({ page, server }) => {
```

Comment:
what happens if the "first page.route" throws", we don't reach the "second page.route"?